core/src/main/java/org/infinispan/notifications/cachelistener/cluster/ClusterListenerReplicateCallable.java

```diff
@@ -99,8 +97,7 @@ public Void call() throws Exception {
                                                                            cacheManagerNotifier, eventManager, sync);
                   // We only need to use the post cache event filter when a filter is provided, since the remote cluster listener
                   // will ignore pre events anyways.  This is mostly so a user filter is not notified of the pre event.
-                  cacheNotifier.addListener(listener,
```

Comment:
Good point. I think my judgement was wrong when removing that ... I think I'll put it back.